### PR TITLE
Fix Flutter tool to use correct adb parameter and collect Realtime value.

### PR DESCRIPTION
**Pull request by terrylucas**
## Description

Fix Flutter tool to pass the correct parameter to adb.  Use  **`_package.id`** instead of **`_package._launchActivity`**.

For example, **`_package._launchActivity`** would return **`io.flutter.demo.gallery/io.flutter.demo.gallery/mainActivity`**

when passed to adb the result is an error in stdout (however, exit code is 0).

The value passed should be **`_package.id`** which would return **'io.flutter.demo.gallery'**.  This will correctly call:

**`shell dumpsys meminfo io.flutter.demo.gallery`**

and stdout will return a properly formed meminfo that will then be parsed correctly.

Added collecting of the Realtime timestamp.

The additional output line parsed, in stdout, would be:

**`Uptime: 441088659 Realtime: 521464097`**

The Realtime is milliseconds since the device was booted (value zero) including deep sleep.
This clock is guaranteed to be monotonic, and continues to tick even in power saving mode. The value zero is Unix Epoch UTC (Jan 1, 1970 00:00:00). This DateTime, from USA PST, would be Dec 31, 1960 16:00:00 (UTC - 8 hours).

## Tests

Added checks to parse the Realtime line in stdout and validate the JSON payload returns the expected Realtime value from the meminfo payload.

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -800,9 +801,18 @@ Map<String, String> parseAdbDeviceProperties(String str) {
...
   final AndroidMemoryInfo androidMemoryInfo = AndroidMemoryInfo();
-  input
-    .split('\n')
+  
+  final lines = input.split('\n');
+
+  var timelineData = lines.firstWhere((String line) =>
+    line.startsWith('${AndroidMemoryInfo._kUpTimeKey}: '));
+  final times = timelineData.trim().split('${AndroidMemoryInfo._kRealTimeKey}:');
+print("=====> ${times.last}");
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Remove prints

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 2]
```diff
@@ -800,9 +801,18 @@ Map<String, String> parseAdbDeviceProperties(String str) {
 /// For more information, see https://developer.android.com/studio/command-line/dumpsys.
 @visibleForTesting
 AndroidMemoryInfo parseMeminfoDump(String input) {
+print("=====> ENTER parseMeminfoDump");
   final AndroidMemoryInfo androidMemoryInfo = AndroidMemoryInfo();
-  input
-    .split('\n')
+  
+  final lines = input.split('\n');
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_device.dart]: To make the lints happy here you'll need to declare the types

**Review by jonahwilliams — APPROVED** [review]
Thanks for fixing this. Overall LGTM

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 1 above)
Comment (terrylucas) [reply packages/flutter_tools/lib/src/android/android_device.dart]: Sorry forgot to hit save. Done.

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 2 above)
Comment (terrylucas) [reply packages/flutter_tools/lib/src/android/android_device.dart]: Same here and a couple of other places too (forgot to his save).  On a different machine w/o auto-save (turned on now ;-).

**Comment by terrylucas** [comment]
Jonah just noticed I hit dartfmt and a bunch of format changes do you folks not use dartfmt?  I can revert the format changes.

**Comment by jonahwilliams** [comment]
Unfortunately we don't :( 

If you don't mind reverting the formatting. I don't think it looks any worse, but it does make the diff significantly bigger than it needs to be

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 3]
```diff
@@ -801,8 +802,15 @@ Map<String, String> parseAdbDeviceProperties(String str) {
...
   final AndroidMemoryInfo androidMemoryInfo = AndroidMemoryInfo();
-  input
-    .split('\n')
+  
+  final List<String> lines = input.split('\n');
+
+  final String timelineData = lines.firstWhere((String line) =>
+    line.startsWith('${AndroidMemoryInfo._kUpTimeKey}: '));
+  final List<String> times = timelineData.trim().split('${AndroidMemoryInfo._kRealTimeKey}:');
+  androidMemoryInfo.realTime = int.tryParse(times.last.trim()) ?? 0;;
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_device.dart]:
```suggestion
  androidMemoryInfo.realTime = int.tryParse(times.last.trim()) ?? 0;
```

**Comment by jonahwilliams** [comment]
I've labelled this so that it will be automatically merged whenever tests finish & the build is green

**Pull request merged**